hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackPlanActionExecutor.java

```diff
@@ -113,6 +113,7 @@ protected Option<HoodieRollbackPlan> requestRollback(String startRollbackTime) {
       HoodieRollbackPlan rollbackPlan = new HoodieRollbackPlan(new HoodieInstantInfo(instantToRollback.getTimestamp(),
           instantToRollback.getAction()), rollbackRequests, LATEST_ROLLBACK_PLAN_VERSION);
       if (!skipTimelinePublish) {
+        table.validateForLatestTimestamp(rollbackInstant.getTimestamp());
```

Comment:
Only the compaction instant timestamp should be kept serial with the regular instant timestamps, the other table actions are okay (for e.g. clustering, rollback, cleaning).

Reply:
not sure how did you arrive at that conclusion. If you look at the scenario called out in the jira, it could be any of write actions (commits, deltacommits, replace commits). may be we don't need rollback, but all writes might def need this fix w/ multi-writer scenarios. for a single writer, anyways txnManager should be a no-op, and so should not have any impact.

Reply:
I have checked https://issues.apache.org/jira/browse/HUDI-7507, it looks like the `TrueTime` API on master can guarantee that each timestamp is the latest (or monotonically increasing), so the main problem here is the time generation in memory and the write of requested file are not put into the same trasanction, can we put them in one transaction just like what we do to the completion time for completed metadata files on master?

Reply:
> Unfortunately (A) has the following drawbacks
Every operation must now hold the table lock when computing its plan even if it's an expensive operation and will take a while
Users of HUDI cannot easily set their own instant time of an operation, and this restriction would break any public APIs that allow this and would require deprecating those APIs.

Even with these drawbacks, I still think A is the right fix, even if we chech the requested instant is always the latest on the timeline, does not mean that instant is following the `TrueTime`(monotonically increasing physically on nodes), and there might still be a use case where the requested instant is the latest but physically the instant happens ealier(because of the clock skew on the machines).

Actually we already bring in completion time based compaction time generation on master (a compaction plan only include files that are finished earlier than the compaction instant), we should also do that for cleaning(and this is the only problem that should fix on master).

Reply:
I think we should guarantee monotonically increasing instant time generation on the timeline on storage as @danny0405 mentioned.  Adding per-action handling for the same guarantee is unmaintainable.

Reply:
Is the check needed only for mor table on 0.x-branch? Because with marker based rollback, we would generate a deterministic rollback plan with the orphan files anyway, can you elaborate the case a little more here?